### PR TITLE
Update: Always fire jetpack_activate_module after update of module.

### PR DESCRIPTION
**Pull request by enejb**
Fixes issue that were created becasue we didn't always call the activate_module action. 

#### Changes proposed in this Pull Request:
- Remove any calls to update_ modules. 

#### Testing instructions:
- Activate, deactivate different modules. 
Does it work as expected?

### CODE REVIEW

File: class.jetpack.php [hunk 1]
```diff
@@ -331,13 +331,70 @@ function plugin_upgrade() {
...
+			foreach( $new_active_modules as $module ) {
+				/**
+				 * Fires when a specific module is activated.
+				 *
+				 * @since 1.9.0
+				 *
+				 * @param string $module Module slug.
+				 * @param boolean $success whether the module was activated. @since 4.2
+				 */
+				do_action( 'jetpack_activate_module', $module, $success );
```
Comment (lezama) [inline class.jetpack.php]: isn't better to sync a single action `jetpack_activate_modules( `$modules` )`?

**Comment by lezama** [comment]
cc @eliorivero for a second pair of eyes

File: sync/class.jetpack-sync-actions.php [hunk 2]
```diff
@@ -164,6 +164,8 @@ static function initialize_sender() {
 }
 
 // Allow other plugins to add filters before we initialize the actions.
-add_action( 'init', array( 'Jetpack_Sync_Actions', 'init' ), 11, 0 );
+// Load the listeners if before modules get loaded so that we can capture version changes etc.
```
Comment (lezama) [inline sync/class.jetpack-sync-actions.php]: extra ` if ` :)

**Comment by lezama** [comment]
so much clean \o/

**Comment by gravityrail** [comment]
OMG so great. Merging.

**Comment by gravityrail** [comment]
Note to @jeherve: I had to exclude core rest api endpoints change when merging to branch-4.2

**Pull request merged**